Assets/Scripts/UI/Tooltip.cs

```diff
@@ -0,0 +1,221 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace UI
+{
+    public class Tooltip : MonoBehaviour
+    {
+        [Tooltip("Highlight color when hover over object.")]
+        public Color highlightColor = new Color(0.38f, 0.97f, 0.44f);
+
+        [Tooltip("Fade speed of the color change (slow -> quick)")]
+        public float highlightSpeed = 4f;
+
+        [Tooltip("Show a text over the interacted object.")]
+        public bool showTooltip = true;
+
+        [Tooltip("Show a predefined UI Panel over the interacted object (intend to use for items in inventory)")]
+        public GameObject tooltipUIPanel;
+
+        [Tooltip("Show the tooltip fixed the object instead of following the mouse.")]
+        public bool fixedToTheObject = true;
```

Comment:
should i just leave this as **true** because by far we only have 2 way of showing tooltip, that are **fixedToTheObject** and **followingMouseCursor**. If bool value is true then style 1, otherwise style 2. So it depends on what we want to define by default

Reply:
Then make it so **followingMouseCursor** can be toggled on but **fixedToTheObject** will be the default

Reply:
i see, will do